beacontools/device_filters.py

```diff
@@ -21,53 +23,80 @@ def matches(self, filter_props):
 
         return found_one
 
+    def __repr__(self):
+        return "{}({})".format(
+            self.__class__.__name__,
+            ", ".join(["=".join((k, str(v),)) for k, v in self.properties.items()]))
+
+
 class IBeaconFilter(DeviceFilter):
     """Filter for iBeacon."""
 
     def __init__(self, uuid=None, major=None, minor=None):
         """Initialize filter."""
         super(IBeaconFilter, self).__init__()
-        if not uuid and not major and not minor:
+        if uuid is None and major is None and minor is None:
             raise ValueError("IBeaconFilter needs at least one argument set")
-        if uuid:
+        if uuid is not None:
             self.properties['uuid'] = uuid
-        if major:
+        if major is not None:
             self.properties['major'] = major
-        if minor:
+        if minor is not None:
             self.properties['minor'] = minor
 
+
 class EddystoneFilter(DeviceFilter):
     """Filter for Eddystone beacons."""
 
     def __init__(self, namespace=None, instance=None):
         """Initialize filter."""
         super(EddystoneFilter, self).__init__()
-        if not namespace and not instance:
+        if namespace is None and instance is None:
             raise ValueError("EddystoneFilter needs at least one argument set")
-        if namespace:
+        if namespace is not None:
             self.properties['namespace'] = namespace
-        if instance:
+        if instance is not None:
             self.properties['instance'] = instance
 
+
 class EstimoteFilter(DeviceFilter):
     """Filter for Estimote beacons."""
 
     def __init__(self, identifier=None, protocol_version=None):
         """Initialize filter."""
         super(EstimoteFilter, self).__init__()
-        if not identifier and not protocol_version:
+        if identifier is None and protocol_version is None:
             raise ValueError("EstimoteFilter needs at least one argument set")
-        if identifier:
+        if identifier is not None:
             self.properties['identifier'] = identifier
-        if protocol_version:
+        if protocol_version is not None:
             self.properties['protocol_version'] = protocol_version
 
+
 class BtAddrFilter(DeviceFilter):
     """Filter by bluetooth address."""
 
+    @staticmethod
+    def parses_invalid_mac(mac):
+        "tries to efficiently parse a MAC Address, returns True on success otherwise False"
+        if mac is None:
+            return True
+        try:
+            mac = mac.replace("-", "").replace(".", "").replace(":", "")
```

Comment:
Instead of doing all these string substitutions and the final length check, it would be more efficient to use a regular expression. This way we can also check that only valid hex digits are used.  I would suggest 
```(?:[0-9a-f]{2}:){5}(?:[0-9a-f]{2})```.

Reply:
I was thinking about using a regex, but my code runs faster, when the mac address is wrong - I'll use the regex :)

Reply:
@citruz okay this is a bit embarrassing - why does the import from utils fail? Please have a look at my current code here: https://github.com/cereal/beacontools/pull/2
Sorry <_<'

Reply:
This is because of a cyclic import: utils -> device_filters -> utils. You need to move the ```from .device_filters import ...``` line from the top to the body of the ```get_mode``` function, similar to ```is_packet_type```.
Another note: At the moment, the MAC address must be lowercase since the output generated by ```bt_addr_to_string``` will always be lowercase. We could convert the address passed to ```BtAddrFilter.__init__``` to lowercase before storing it, this way the user can use lower, upper or mixed-case addresses. The regex can be case-insensitive then.

Reply:
hah! That was easy! :P 
Okay, I'll change it to a regex without any flag set and change it in `BtAddrFilter.__init__` :)

Reply:
okay I pushed :+1: